src/egeoip_db_updater.erl

```diff
@@ -0,0 +1,100 @@
+-module(egeoip_db_updater).
+-behaviour(gen_server).
+
+%% ------------------------------------------------------------------
+%% API Function Exports
+%% ------------------------------------------------------------------
+
+-export([start_link/1]).
+
+%% ------------------------------------------------------------------
+%% gen_server Function Exports
+%% ------------------------------------------------------------------
+
+-export([init/1, terminate/2, code_change/3]).
+-export([handle_call/3, handle_cast/2, handle_info/2]).
+
+%% ------------------------------------------------------------------
+%% Macro Definitions
+%% ------------------------------------------------------------------
+
+-define(CB_MODULE, ?MODULE).
+-define(REFRESH_INTERVAL_MS, timer:hours(12)).
+-define(REFRESH_TIMEOUT_MS, timer:seconds(15)).
+
+%% ------------------------------------------------------------------
+%% Record Definitions
+%% ------------------------------------------------------------------
+
+-record(state, {
+    dbpath :: string(),
+    db :: binary()
+}).
+-type state() :: #state{}.
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+-spec start_link(DbPath :: undefined
+                         | string())
+      -> ignore | {ok, pid()}.
+start_link(undefined) -> ignore;
+start_link(DbPath) -> gen_server:start_link(?CB_MODULE, [DbPath], []).
+
+%% ------------------------------------------------------------------
+%% gen_server Function Definitions
+%% ------------------------------------------------------------------
+
+-spec init([DbPath :: string()])
+      -> {ok, State :: state()}.
+init([DbPath]) ->
+    erlang:send_after(0, self(), reload_db),
+    {ok, #state{ dbpath = DbPath }}.
+
+-spec terminate(Reason :: term(),
+                State :: state())
+      -> ok.
+terminate(_Reason, _State) ->
+    ok.
+
+-spec code_change(OldVsn :: term(),
+                  State :: state(),
+                  Extra :: term())
+      -> {ok, NewState :: state()}.
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+-spec handle_call(Request :: term(),
+                  From :: {Pid :: pid(), Tag :: binary()},
+                  State :: state())
+      -> {noreply, State :: state()}
+       | {reply, Reply :: term(), State :: state()}.
+handle_call(_Request, _From, State) ->
+    {noreply, State}.
+
+-spec handle_cast(Request :: term(),
+                  State :: state())
+      -> {noreply, NewState :: state()}.
+handle_cast(_Request, State) ->
+    {noreply, State}.
+
+-spec handle_info(Info :: reload
+                        | term(),
+                  State :: state())
+      -> {noreply, NewState :: state()}.
+handle_info(reload_db, #state{ dbpath = DbPath, db = Db } = State) ->
+    NewState
+        = case httpc:request(get, {DbPath, []}, [{timeout, ?REFRESH_TIMEOUT_MS}], []) of
```

Comment:
Sounds good. Their hosts will happily comply (or cloudflare hosts, for that matter).

For a date before the last modified date:

```bash
⇒  curl -I https://geolite.maxmind.com/download/geoip/database/GeoLiteCountry/GeoIP.dat.gz -H 'If-Modified-Since: Tue, 04 Jul 2017 14:47:51 GMT'
HTTP/2 200
date: Mon, 31 Jul 2017 17:14:03 GMT
content-type: application/octet-stream
content-length: 640135
set-cookie: __cfduid=ddf44648f70fbd724b17657930c2758441501521243; expires=Tue, 31-Jul-18 17:14:03 GMT; path=/; domain=.maxmind.com; HttpOnly
last-modified: Tue, 04 Jul 2017 14:47:52 GMT
etag: "595baa98-9c487"
cf-cache-status: HIT
expires: Mon, 31 Jul 2017 21:14:03 GMT
cache-control: public, max-age=14400
accept-ranges: bytes
server: cloudflare-nginx
cf-ray: 38723119bf846206-LIS
```

For a date after the last modified date:
```bash
⇒  curl -I https://geolite.maxmind.com/download/geoip/database/GeoLiteCountry/GeoIP.dat.gz -H 'If-Modified-Since: Tue, 04 Jul 2017 14:47:52 GMT'
HTTP/2 304
date: Mon, 31 Jul 2017 17:13:58 GMT
set-cookie: __cfduid=d300a40a861d11fdabf78166ea98c2c8d1501521238; expires=Tue, 31-Jul-18 17:13:58 GMT; path=/; domain=.maxmind.com; HttpOnly
last-modified: Tue, 04 Jul 2017 14:47:52 GMT
etag: "595baa98-9c487"
cf-cache-status: HIT
expires: Mon, 31 Jul 2017 21:13:58 GMT
cache-control: public, max-age=14400
server: cloudflare-nginx
cf-ray: 387230ff4f146212-LIS
```

What would the time be though?